src/auth.ts

```diff
@@ -16,6 +17,10 @@ export async function configureAuthentication() {
   const id = core.getInput(constants.INPUT_SERVER_ID);
   const username = core.getInput(constants.INPUT_SERVER_USERNAME);
   const password = core.getInput(constants.INPUT_SERVER_PASSWORD);
+  const settingsDirectory =
+    core.getInput(constants.INPUT_SETTINGS_PATH) || path.join(os.homedir(), M2_DIR);
+  // Consider setting overwriteSettings = false by default in next major release
+  const overwriteSettings = getBooleanInput(constants.INPUT_OVERWRITE_SETTINGS, true);
```

Comment:
As I see, it always returns `true` if the second parameter is `true`, for any inputs, is it expected? or I miss something?